File: tests/rspec/pages/site/site_form.rb
class SiteForm < ControlPanelPage
  set_url_matcher /msm\/edit/

  element :settings_icon, '.section-header__options a.icon--settings'
  element :title, '.section-header__title'
  element :add_site_button, '.section-header__controls a.btn.action'

  element :save_button, '.form-btns-top input.btn'
  element :name, 'input[name="site_label"]'
  element :short_name, 'input[name="site_name"]'
  element :online, 'a[data-toggle-for="is_site_on"]'
  element :description, 'textarea[name="site_description"]'

  def add_site(options = {})
    defaults = {
      name: 'Rspec Site',
      short_name: 'rspec_site',
      description: 'Lorem ipsum...'
    }

    options = defaults.merge(options)

    name.set options[:name]
    short_name.set options[:short_name]
    description.set options[:description]
    save_button.click
  end

  def load_edit_for_site(id)
    self.open_dev_menu
    click_link 'Site Manager'
    find('a[href*="cp/msm/edit/' + id + '"]').click
  end
end

File: tests/cypress/cypress/elements/pages/site/SiteForm.js
import ControlPanel from '../ControlPanel'

class SiteForm extends ControlPanel {
  constructor() {
    super()
    this.urlMatch = /msm\/edit/;
    this.elements({

      'settings_icon': '.section-header__options a.icon--settings',
      'title': '.section-header__title',
      'add_site_button': '.main-nav__toolbar a.button--action',

      'save_button': 'form .form-btns-top button[type=submit][value="save_and_close"]',
      'name': 'input[type!=hidden][name="site_label"]',
      'short_name': 'input[type!=hidden][name="site_name"]',
      'online': '[data-toggle-for="is_site_on"]',
      'description': 'textarea[name="site_description"]'

    })
  }

  add_site(passed_options = {}) {
    const defaults = {
      name: 'Rspec Site',
      short_name: 'rspec_site',
      description: 'Lorem ipsum...'
    }

    const options = Cypress._.merge(defaults, passed_options)

    this.get('name').type(options.name)
    this.get('short_name').clear().type(options.short_name)
    this.get('description').type(options.description)
    cy.get('.form-btns-top .saving-options').click()
    this.get('save_button').click()
  }

  load_edit_for_site(id) {
    this.open_dev_menu()
    this.get('main_menu').find('a:contains("Site Manager")').click()
    this.get('wrap').find('a[href*="cp/msm/edit/' + id + '"]').click()
  }
}
export default SiteForm;